app/src/classic/java/org.stepic.droid/ui/activities/LaunchActivity.kt

```diff
@@ -172,34 +179,47 @@ class LaunchActivity : BackToExitActivityBase(), LoginView {
         }
     }
 
-    private fun initSocialRecycler(googleApiClient: GoogleApiClient?) {
-        val pixelForPadding = DpPixelsHelper.convertDpToPixel(4f, this)//pixelForPadding * (count+1)
-        val widthOfItem = resources.getDimension(R.dimen.height_of_social)//width == height
-        val count = SocialManager.SocialType.values().size
-        val widthOfAllItems = widthOfItem * count + pixelForPadding * (count + 1)
+    private fun initSocialRecycler(googleApiClient: GoogleApiClient?, state: SocialAuthAdapter.State = SocialAuthAdapter.State.NORMAL) {
+        socialListRecyclerView.layoutManager = GridLayoutManager(this, 3)
 
-        val display = windowManager.defaultDisplay
-        val size = Point()
-        display.getSize(size)
-        val widthOfScreen = size.x
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
+            socialListRecyclerView.itemAnimator = FadeInDownAnimator()
+            socialListRecyclerView.itemAnimator.removeDuration = 0
```

Comment:
collapse animation is not working

Reply:
collapse animation not working properly so I disabled it